From 8c52fe555aab7f2ae869b8a39802872b066942f6 Mon Sep 17 00:00:00 2001
From: Jan Vrany <jan.vrany@fit.cvut.cz>
Date: Fri, 31 Jul 2020 17:54:06 +0000
Subject: [PATCH] Add -Wno-stringop-overflow when compiling with GCC 10

Starting with GCC 10, some code fails to compile with -Wall -Werror,
failing after (possibly false) warning about possible string overflow.

This commit add -Wno-stringop-overflow when compiling with GCC 10.

Signed-off-by: Jan Vrany <jan.vrany@fit.cvut.cz>
---
 omrmakefiles/rules.linux.mk | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/omrmakefiles/rules.linux.mk b/omrmakefiles/rules.linux.mk
index 1332f54be..6a58aa263 100644
--- a/omrmakefiles/rules.linux.mk
+++ b/omrmakefiles/rules.linux.mk
@@ -37,6 +37,12 @@ ifeq (gcc,$(OMR_TOOLCHAIN))
     GLOBAL_CFLAGS+=-MMD -MP
     GLOBAL_CXXFLAGS+=-MMD -MP
   endif
+
+  ifeq ($(shell expr `($(CC) -dumpversion 2>/dev/null || echo 1) | cut -d. -f1` \>= 10),1)
+    # Suppress stringop-overflow warning on GCC >=10
+    GLOBAL_CFLAGS+=-Wno-stringop-overflow
+    GLOBAL_CXXFLAGS+=-Wno-stringop-overflow
+  endif
 endif
 
 ###
-- 
2.27.0

